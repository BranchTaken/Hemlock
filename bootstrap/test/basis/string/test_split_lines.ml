open! Basis.Rudiments
open! Basis
open String
open Format

let test () =
  let test_split_lines s = begin
    printf "split_lines %S -> [" s;
    List.iteri (split_lines s)~f:(fun i substr ->
      if Uns.(i > 0L) then printf "; ";
      printf "%S" substr
    );
    printf "]\n";

    printf "split_lines_rev %S -> [" s;
    List.iteri (split_lines_rev s)~f:(fun i substr ->
      if Uns.(i > 0L) then printf "; ";
      printf "%S" substr
    );
    printf "]\n";
  end in
  test_split_lines "ab";

  test_split_lines "\nab";
  test_split_lines "a\nb";
  test_split_lines "ab\n";
  test_split_lines "\na\nb\n";

  test_split_lines "\r\nab";
  test_split_lines "a\r\nb";
  test_split_lines "ab\r\n";
  test_split_lines "\r\na\r\nb\r\n";

  test_split_lines "a\r\r\nb";

  test_split_lines "a\n\nb";
  test_split_lines "a\r\n\r\nb";

  test_split_lines "a\n\r\nb";
  test_split_lines "a\r\n\nb";

  test_split_lines "a\n\n\nb"

let _ = test ()
