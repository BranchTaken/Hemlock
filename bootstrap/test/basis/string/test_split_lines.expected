split_lines "ab" -> ["ab"]
split_lines_rev "ab" -> ["ab"]
split_lines "\nab" -> [""; "ab"]
split_lines_rev "\nab" -> ["ab"; ""]
split_lines "a\nb" -> ["a"; "b"]
split_lines_rev "a\nb" -> ["b"; "a"]
split_lines "ab\n" -> ["ab"; ""]
split_lines_rev "ab\n" -> [""; "ab"]
split_lines "\na\nb\n" -> [""; "a"; "b"; ""]
split_lines_rev "\na\nb\n" -> [""; "b"; "a"; ""]
split_lines "\r\nab" -> [""; "ab"]
split_lines_rev "\r\nab" -> ["ab"; ""]
split_lines "a\r\nb" -> ["a"; "b"]
split_lines_rev "a\r\nb" -> ["b"; "a"]
split_lines "ab\r\n" -> ["ab"; ""]
split_lines_rev "ab\r\n" -> [""; "ab"]
split_lines "\r\na\r\nb\r\n" -> [""; "a"; "b"; ""]
split_lines_rev "\r\na\r\nb\r\n" -> [""; "b"; "a"; ""]
split_lines "a\r\r\nb" -> ["a\r"; "b"]
split_lines_rev "a\r\r\nb" -> ["b"; "a\r"]
split_lines "a\n\nb" -> ["a"; ""; "b"]
split_lines_rev "a\n\nb" -> ["b"; ""; "a"]
split_lines "a\r\n\r\nb" -> ["a"; ""; "b"]
split_lines_rev "a\r\n\r\nb" -> ["b"; ""; "a"]
split_lines "a\n\r\nb" -> ["a"; ""; "b"]
split_lines_rev "a\n\r\nb" -> ["b"; ""; "a"]
split_lines "a\r\n\nb" -> ["a"; ""; "b"]
split_lines_rev "a\r\n\nb" -> ["b"; ""; "a"]
split_lines "a\n\n\nb" -> ["a"; ""; ""; "b"]
split_lines_rev "a\n\n\nb" -> ["b"; ""; ""; "a"]
